apps/meteor/client/views/navigation/sidepanel/SidepanelItem/useParentTeamData.ts

```diff
@@ -0,0 +1,61 @@
+import type { ITeam } from '@rocket.chat/core-typings';
+import { TEAM_TYPE } from '@rocket.chat/core-typings';
+import { useEndpoint, useUserId } from '@rocket.chat/ui-contexts';
+import { useQuery, keepPreviousData } from '@tanstack/react-query';
+
+import { goToRoomById } from '../../../../lib/utils/goToRoomById';
+
+type APIErrorResult = { success: boolean; error: string };
+
+export const useParentTeamData = (teamId?: ITeam['_id']) => {
+	const userId = useUserId();
+
+	if (!teamId) {
+		throw new Error('invalid rid');
+	}
+
+	if (!userId) {
+		throw new Error('invalid uid');
+	}
+
+	const teamsInfoEndpoint = useEndpoint('GET', '/v1/teams.info');
+	const userTeamsListEndpoint = useEndpoint('GET', '/v1/users.listTeams');
+
+	const {
+		data: teamInfoData,
+		isLoading: teamInfoLoading,
+		isError: teamInfoError,
+	} = useQuery({
+		queryKey: ['teamId', teamId],
+		queryFn: async () => teamsInfoEndpoint({ teamId }),
+		placeholderData: keepPreviousData,
+		retry: (_, error: APIErrorResult) => error?.error === 'unauthorized' && false,
+	});
+
+	const { data: userTeams, isLoading: userTeamsLoading } = useQuery({
+		queryKey: ['userId', userId],
```

Comment:
do we have this keys in the [queryKeys](https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/client/lib/queryKeys.ts) file? If not, might be a good idea to add there.